### PR TITLE
External script support for CSP

### PR DESCRIPTION
**Pull request by sunecosuri**
refs : https://github.com/nuxt/nuxt.js/issues/2606

This PR is Changed to allow external scripts for content security policy

e.g.) `script-src  'self' sha256-xKS1DGTF21G4Kpdsfaadsx3b+dw= https://example.com`

### CODE REVIEW

File: test/basic.ssr.test.js [hunk 1]
```diff
@@ -24,7 +24,10 @@ test.serial('Init Nuxt.js', async t => {
       stats: false
     },
     render: {
-      csp: true
+      csp: {
+        hashAlgorithm: 'sha256',
+        allowDomains: ['https://example.com', 'https://example.io']
```
Comment (pi0) [inline test/basic.ssr.test.js]: Please set  `allowedDomains` in `common/options.js` with an empty array to indicate such options is available

File: lib/core/middleware/nuxt.js [hunk 2]
```diff
@@ -68,9 +68,13 @@ module.exports = async function nuxtMiddleware(req, res, next) {
     }
 
     if (this.options.render.csp) {
+      let allowSources = cspScriptSrcHashes
```
Comment (pi0) [inline lib/core/middleware/nuxt.js]: `let allowedSources = cspScriptSrcHashes.concat(this.options.render.csp.allowDomains)`

**Review by pi0 — COMMENTED** [review]
Thanks for the PR. Isn't it better to use `Allowed*` instead of `Allow*`? Also it may not be limited to domains only so maybe using `Sources` in all places instead of `Domains`?

File: test/basic.ssr.test.js (hunk 1 above)
Comment (sunecosuri) [reply test/basic.ssr.test.js]: Thank you for review!
I changed to the contents for the review, is it okay so?

Comment (pi0) [reply test/basic.ssr.test.js]: Yes. Thanks :)

**Pull request merged**